code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/AEPPushTemplate.kt

```diff
@@ -0,0 +1,400 @@
+
+    // Optional, small icon for the notification
+    internal var smallIcon: String?
+        private set
+
+    // Optional, large icon for the notification
+    internal var largeIcon: String?
+        private set
+
+    // Optional, image to show in the notification
+    internal var imageUrl: String?
+        private set
+
+    // Optional, action type for the notification
+    private var actionType: ActionType?
+
+    // Optional, action uri for the notification
+    internal var actionUri: String?
+        private set
+
+    // Optional, action buttons for the notification
+    internal var actionButtonsString: String?
+        private set
+
+    // Optional, Body of the message shown in the expanded message layout (setCustomBigContentView)
+    internal var expandedBodyText: String?
+        private set
+
+    // Optional, Text color for adb_body. Represented as six character hex, e.g. 00FF00
+    internal var expandedBodyTextColor: String?
+        private set
+
+    // Optional, Text color for adb_title. Represented as six character hex, e.g. 00FF00
+    internal var titleTextColor: String?
+        private set
+
+    // Optional, Color for the notification's small icon. Represented as six character hex, e.g.
+    // 00FF00
+    internal var smallIconColor: String?
+        private set
+
+    // Optional, Color for the notification's background. Represented as six character hex, e.g.
+    // 00FF00
+    internal var notificationBackgroundColor: String?
+        private set
+
+    // Optional, If present and a notification with the same tag is already being shown, the new
+    // notification replaces the existing one in the notification drawer.
+    internal var tag: String?
+        private set
+
+    // Optional, If present sets the "ticker" text, which is sent to accessibility services.
+    internal var ticker: String?
+        private set
+
+    // Optional, the type of push template this payload contains
+    internal var templateType: PushTemplateType?
+        private set
+
+    // Optional, when set to false or unset, the notification is automatically dismissed when the
+    // user clicks it in the panel. When set to true, the notification persists even when the user
+    // clicks it.
+    internal var isNotificationSticky: Boolean?
+        private set
+
+    // flag to denote if the PushTemplate was built from an intent
+    internal var isFromIntent: Boolean?
+
+    /**
+     * Constructor to create a push template object from the data payload.
+     *
+     * @param data [Map] of key-value pairs representing the push template data payload
+     */
+    constructor(data: Map<String, String>?) {
+        // fast fail (via IllegalArgumentException) if required data is not present
+        if (data.isNullOrEmpty()) throw IllegalArgumentException("Push template data is null.")
+        this.messageData = data.toMutableMap()
+        val title = DataReader.getString(data, PushTemplateConstants.PushPayloadKeys.TITLE)
```

Comment:
`DataReader.getString()` will throw a `DataReaderException` if the key is not found. This exception does not make much sense to the caller because they do not need to be aware of how the key reading happens in this constructor.  Can you use `DataReader.optString` with a null fallback and then finally throw an `IllegalArgumentException`. 
Same comment for all other cases where `DataReader.get*` is being used.